pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/operands/FilterOperand.java

```diff
@@ -230,6 +230,7 @@ public Predicate(List<RexExpression> functionOperands, DataSchema dataSchema) {
           "Expected 2 function ops for Predicate but got:" + functionOperands.size());
       _lhs = TransformOperand.toTransformOperand(functionOperands.get(0), dataSchema);
       _rhs = TransformOperand.toTransformOperand(functionOperands.get(1), dataSchema);
+      // TODO: correctly handle type hoisting. here we favor left type first. which is not correct for say 2 < 2.3
```

Comment:
will fix in follow up PR. basically `WHERE 2 < 2.3` is false and `WHERE 2.3 > 2` is true nowadays